### PR TITLE
feat: dynamically generate for orbit chains 

### PR DESCRIPTION
**Pull request by chrstph-dvx**

### CODE REVIEW

File: .github/scripts/generateMatrix.ts [hunk 1]
```diff
@@ -91,35 +98,68 @@ const arbitrumCommands: Command[] = [
...
 
-const matrix: Record<'include', Command[]> = {
-  include: arbitrumCommands.concat(orbitCommands),
-};
+const l1ChainIds = [1, 11155111, 17000]; // Mainnet, sepolia, holesky
+(async () => {
+  for (let { name, chainID, partnerChainID } of customNetworks) {
+    // For Orbit chain settling on L1, use uniswap as source. Otherwise use arbified token list
+    const inputUniswapTokenList = l1ChainIds.includes(partnerChainID)
+      ? 'https://tokens.uniswap.org'
```
Comment (fionnachan) [inline .github/scripts/generateMatrix.ts:135]: we would also want to use uniswap's list for tokens on Base for Orbit chains on top of Base

File: .github/scripts/fetchOrbitChainsData.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  const result: L2NetworkWithRpc[] = orbitChains.map((orbitChain) =>
+    parseChainToL2Network(orbitChain),
+  );
+
+  fs.writeFileSync(
+    fileName,
+    JSON.stringify({
+      data: result,
+    }),
+  );
```
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply .github/scripts/fetchOrbitChainsData.ts]: Pretty much yes

File: src/scripts/fetchOrbitChainsData.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+function parseChainToL2Network({
+  chainId,
+  nativeToken,
+  isTestnet,
+  slug,
+  bridgeUiConfig,
+  parentChainId,
+  tokenBridge,
+  ...chain
```
Comment (spsjvc) [inline src/scripts/fetchOrbitChainsData.ts:79]: hm in this situation I'd maybe not do spread rest, to make it clearer what data is being mapped

File: .env.sample [hunk 4]
```diff
@@ -3,6 +3,10 @@ ARB_ONE_RPC="https://arbitrum.infura.io/v3/${INFURA_KEY}"
 ARB_SEPOLIA_RPC="https://arbitrum-sepolia.infura.io/v3/${INFURA_KEY}"
 MAINNET_RPC="https://mainnet.infura.io/v3/${INFURA_KEY}"
 SEPOLIA_RPC="https://sepolia.infura.io/v3/${INFURA_KEY}"
+HOLESKY_RPC="https://holesky.infura.io/v3/${INFURA_KEY}"
+ARB_NOVA_RPC="https://nova.arbitrum.io/rpc"
+BASE_RPC="https://mainnet.base.org"
+BASE_SEPOLIA_RPC="https://sepolia.base.org"
```
Comment (spsjvc) [inline .env.sample:9]: we can use infura for everything except Arbitrum Nova

File: .github/actions/generate-orbit-chains-data/action.yml
Comment (spsjvc) [inline .github/actions/generate-orbit-chains-data/action.yml:1]: are these actions necessary?

```
- ./.github/actions/generate-orbit-chains-data
- ./.github/actions/restore-orbit-chains-data
```

I feel like they add extra complexity where the only thing you need to do is download a file before each job which can just be `yarn generate:orbitChainsData`

**Review by spsjvc — APPROVED** [review]

**Pull request merged**